File: Week 1 - mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int pyramidHeight;
    int numOfBlocks = 1;    //start at 1, because there's always exactly one block on each side of the top level of the pyramid


    //get height from user input
    do
    {
        pyramidHeight = get_int("Height: ");
    }
    while (pyramidHeight < 1 || pyramidHeight > 8);


    //print the pyramid
    for (int i = 0; i < pyramidHeight; i++)
    {
        //print left side of row
        for (int j = 0; j < pyramidHeight; j++)
        {
            if (j >= pyramidHeight - numOfBlocks)
            {
                printf("#");
            }
            else
            {
                printf(" ");
            }
        }


        //add spaces in the middle
        printf("  ");


        //print right side of row
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        //increment numOfBlocks for next line, then start a new line
        numOfBlocks++;
        printf("\n");
    }
}

File: Week 6 - sentimental-mario-more/mario.py
# prompt user for height of pyramid
print("Height: ", end="")

# continue prompting user until valid input is recieved
while True:
    try:
        height = int(input())

        if height <= 0 or height > 8:
            print("Height: ", end="")
        else:
            break
    except ValueError:
        print("Height: ", end="")

# number of spaces to print on each line before the hash signs start
spaces = height - 1

# for each row in the pyramid
for i in range(height):

    # print the left half
    for j in range(height):
        if j < spaces:
            print(" ", end="")
        else:
            print("#", end="")

    # add two spaces in between halves
    print("  ", end="")

    # print the right half
    for j in range(height - spaces):
        print("#", end="")

    # decrement spaces and start a new line
    spaces -= 1
    print()

